aws_lambda_decorators/utils.py

```diff
@@ -91,13 +94,18 @@ def find_key_case_insensitive(key_name, the_dict):
         The found key name in its original case, if found. Otherwise, returns the searching key name
 
     """
+    desensitise = lambda key: normalise('NFC', key).casefold()
+
+    key_name_lower = desensitise(key_name)
```

Comment:
we could just have the desensitise call in the if conditon

Reply:
We could, but then the `key_name` would be lower cased for every key in the dictionary until a match is found.
I think it makes sense to desensitise the input once and re-use that value.